core/src/main/java/org/elasticsearch/action/support/replication/ReplicationRequest.java

```diff
@@ -171,12 +172,12 @@ long routedBasedOnClusterVersion() {
     }
 
     /** returns the primary term active at the time the operation was performed on the primary shard */
-    public long primaryTerm() {
+    long primaryTerm() {
         return primaryTerm;
     }
 
     /** marks the primary term in which the operation was performed */
-    public void primaryTerm(long term) {
+    void primaryTerm(long term) {
         primaryTerm = term;
```

Comment:
can we assert we can only set it if it's zero and if it's not that it's equal to the parameter?